src/XMLimport.cpp

```diff
@@ -611,144 +611,112 @@ void XMLimport::readHelpPackage()
 void XMLimport::readUnknownPackage()
 {
     while (!atEnd()) {
-        readNext();
+        skipCurrentElement();
```

Comment:
I think this has to be *after* the debug message - as it advances the parser off of the *current* element which you are skipping and leaves it with the parent item being the current one so the details printed out would not be what you think it is...

TBH I have had fun with the XML reader in the past - and the best way to work out what is happening is to single step through the loop in the debugger with watches on `name()`, `attributes()`, `text()` and `tokenType()` whilst also having the XML file open in a suitable editor/viewer...

Reply:
I've checked. This is correct.
```xml
<unknown>
  <inner></inner>
</unknown>
```

In this case we're outputting `unknow`, not `inner` and that's desired.